From d0330234d96ec4eb5376aaa149a1c8fa55a3baaa Mon Sep 17 00:00:00 2001
From: Fabio Bertinatto <fbertina@redhat.com>
Date: Fri, 5 Jan 2024 09:19:59 -0300
Subject: [PATCH] UPSTERAM: <drop>: Revert "use ctx withoutcancel"

This reverts commit 6d92b330e31fc580c73c31f1f82eacdb4c9bd2bd.
---
 pkg/kubelet/kubelet.go | 6 ++++--
 1 file changed, 4 insertions(+), 2 deletions(-)

diff --git a/pkg/kubelet/kubelet.go b/pkg/kubelet/kubelet.go
index 441158250db..3aa9e0d1c4f 100644
--- a/pkg/kubelet/kubelet.go
+++ b/pkg/kubelet/kubelet.go
@@ -1964,8 +1964,10 @@ func (kl *Kubelet) SyncPod(ctx context.Context, updateType kubetypes.SyncPodType
 	// the context for SyncPod is a known and deliberate error, not a generic error.
 	// Use WithoutCancel instead of a new context.TODO() to propagate trace context
 	// Call the container runtime's SyncPod callback
-	sctx := context.WithoutCancel(ctx)
-	result := kl.containerRuntime.SyncPod(sctx, pod, podStatus, pullSecrets, kl.backOff)
+	result := kl.containerRuntime.SyncPod(ctx, pod, podStatus, pullSecrets, kl.backOff)
+	if wait.Interrupted(result.Error()) {
+		return false, err
+	}
 	kl.reasonCache.Update(pod.UID, result)
 	if err := result.Error(); err != nil {
 		// Do not return error if the only failures were pods in backoff
-- 
2.43.0

